airflow/providers/amazon/aws/executors/batch/batch_executor.py

```diff
@@ -0,0 +1,386 @@
+        check_health = conf.getboolean(
+            CONFIG_GROUP_NAME, AllBatchConfigKeys.CHECK_HEALTH_ON_STARTUP, fallback=False
+        )
+
+        if not check_health:
+            return
+
+        self.log.info("Starting Batch Executor and determining health...")
+        try:
+            self.check_health()
+        except AirflowException:
+            self.log.error("Stopping the Airflow Scheduler from starting until the issue is resolved.")
+            raise
+
+    def load_batch_connection(self, check_connection: bool = True):
+        self.log.info("Loading Connection information")
+        aws_conn_id = conf.get(
+            CONFIG_GROUP_NAME,
+            AllBatchConfigKeys.AWS_CONN_ID,
+            fallback=CONFIG_DEFAULTS[AllBatchConfigKeys.AWS_CONN_ID],
+        )
+        region_name = conf.get(CONFIG_GROUP_NAME, AllBatchConfigKeys.REGION_NAME)
+        self.batch = BatchClientHook(aws_conn_id=aws_conn_id, region_name=region_name).conn
+        self.attempts_since_last_successful_connection += 1
+        self.last_connection_reload = timezone.utcnow()
+
+        if check_connection:
+            self.check_health()
+            self.attempts_since_last_successful_connection = 0
+
+    def sync(self):
+        """Sync will get called periodically by the heartbeat method in the scheduler."""
+        if not self.IS_BOTO_CONNECTION_HEALTHY:
+            exponential_backoff_retry(
+                self.last_connection_reload,
+                self.attempts_since_last_successful_connection,
+                self.load_batch_connection,
+            )
+            if not self.IS_BOTO_CONNECTION_HEALTHY:
+                return
+        try:
+            self.sync_running_jobs()
+            self.attempt_submit_jobs()
+        except (ClientError, NoCredentialsError) as error:
+            error_code = error.response["Error"]["Code"]
+            if error_code in INVALID_CREDENTIALS_EXCEPTIONS:
+                self.IS_BOTO_CONNECTION_HEALTHY = False
+                self.log.warning(
+                    f"AWS credentials are either missing or expired: {error}.\nRetrying connection"
+                )
+        except Exception:
+            # We catch any and all exceptions because otherwise they would bubble
+            # up and kill the scheduler process
+            self.log.exception("Failed to sync %s", self.__class__.__name__)
+
+    def sync_running_jobs(self):
+        all_job_ids = self.active_workers.get_all_jobs()
+        if not all_job_ids:
+            self.log.debug("No active Airflow tasks, skipping sync")
+            return
+        describe_job_response = self._describe_jobs(all_job_ids)
+
+        self.log.debug("Active Workers: %s", describe_job_response)
+
+        for job in describe_job_response:
+            if job.get_job_state() == State.FAILED:
+                self._handle_failed_job(job)
+            elif job.get_job_state() == State.SUCCESS:
```

Comment:
There shouldn't be any other relevant states. `get_job_state` can return `State.QUEUED`, `State.RUNNING`, `State.FAILED`, or `State.SUCCESS`. `State.QUEUED` is kind of a catchall for various intermediate states. For both the QUEUED and RUNNING states, we don't want to do anything, so that only leaves the 2 states handled above.